connection.go

```diff
@@ -302,6 +306,13 @@ func (ch *Channel) newConnection(conn net.Conn, initialID uint32, outboundHP str
 	log = log.WithFields(LogField{"connectionDirection", connDirection})
 	peerInfo := ch.PeerInfo()
 
+	sendBufferSize := opts.SendBufferSize
+	for processNamePrefix, override := range opts.SendBufferSizeOverrides {
```

Comment:
move this logic into a separate function (can be on the options type) that takes in the `remotePeer`, ideally this should just be `sendBufferSize := opts.GetSendBufferSize(remotePeer)` (and we could inline that into the `sendCh` constructor too)

Reply:
Once you make it a function on the options, you can also write a unit test that is more functional (but still keep the end-to-end test with a couple of cases)

Reply:
Not sure why `connection_test.go` belongs to a different package -- this means `GetSendBufferSize` has to be exported for unit tests. Feels cleaner unit tests is a small gain compared to adding the downside of unnecessary public API? (But yea pulled the logic into its own function.

Reply:
Separate package is used for the end-to-end test helpers (`testutils` package has helpers for creating channels and test verification, which we are used by all subpackages, but we don't want to export as part of this package's API)

If you want unit tests, you can use `connection_internal_test.go` which is in the same package for "internal" test that need access to the unexported API.

Reply:
Looks like moving to `connection_internal_test.go` will introduce circular import. I'll just leave it as is unless there's a strong opinion on unit tests vs e2e tests.